commandlineparser/include/Parser.h

```diff
@@ -62,6 +64,48 @@ namespace cmdParser
 		//storing the tokenized data.		
 		std::vector<std::string> tokenized_data;
 	};
+
+	template <typename T>
+	void cmdParser::Parser::AddOptions(std::string short_command, std::string long_command, std::string short_description, std::string long_description, T def_value)
+	{
+		static_assert(!std::is_convertible<T, const char *>::value, "const char * data type for string literal is not supported in this version !");
+		
+		std::shared_ptr<cmdParser::Options> obj = std::make_shared <cmdParser::Options_with_value<T>>(short_command, long_command, short_description, long_description, def_value);
+
+		bool is_short_empty = false;
+
```

Comment:
Add a private function  add_option_objects and move the following code to that add function. 

private : 
add_option_object( std::shared_ptr<cmdParser::Options> obj)

Reply:
done.